### PR TITLE
main/sdl2: move to CMake and enable Wayland support

### PR DESCRIPTION
**Pull request by PureTryOut**

### CODE REVIEW

File: main/sdl2/APKBUILD [hunk 1]
```diff
@@ -2,45 +2,55 @@
 # Maintainer: August Klein <amatcoder@gmail.com>
 pkgname=sdl2
 pkgver=2.0.9
-pkgrel=0
+pkgrel=1
 pkgdesc="A development library designed to provide low level access to audio, keyboard, mouse, joystick and graphics"
 url="http://www.libsdl.org"
```
Comment (maxice8) [inline main/sdl2/APKBUILD]: https://www.libsdl.org/

File: main/sdl2/APKBUILD [hunk 2]
```diff
@@ -2,45 +2,55 @@
...
 arch="all"
 license="zlib"
```
Comment (maxice8) [inline main/sdl2/APKBUILD]: Zlib

File: main/sdl2/APKBUILD [hunk 3]
```diff
@@ -2,45 +2,55 @@
...
+		-DRPATH=OFF \
+		-DVIDEO_X11=ON \
+		-DVIDEO_DIRECTFB=ON \
+		-DVIDEO_WAYLAND=ON
 	make
+
+	chmod +x sdl2-config
 }
 
 check() {
```
Comment (maxice8) [inline main/sdl2/APKBUILD]: I think we should use actual upstream testsuite or disable check altogether

**Comment by ncopa** [comment]
Seem like this change broke qemu build:
```
In file included from /home/ncopa/aports/main/qemu/src/qemu-4.0.0/include/ui/sdl2.h:8,
                 from /home/ncopa/aports/main/qemu/src/qemu-4.0.0/ui/sdl2-2d.c:30:
/usr/include/SDL2/SDL_syswm.h:77:10: fatal error: directfb.h: No such file or directory
 #include <directfb.h>
          ^~~~~~~~~~~~
compilation terminated.
make: *** [/home/ncopa/aports/main/qemu/src/qemu-4.0.0/rules.mak:69: ui/sdl2-2d.o] Error 1
make: *** Waiting for unfinished jobs....
In file included from /home/ncopa/aports/main/qemu/src/qemu-4.0.0/include/ui/sdl2.h:8,
                 from /home/ncopa/aports/main/qemu/src/qemu-4.0.0/ui/sdl2-gl.c:32:
/usr/include/SDL2/SDL_syswm.h:77:10: fatal error: directfb.h: No such file or directory
 #include <directfb.h>
          ^~~~~~~~~~~~
compilation terminated.
make: *** [/home/ncopa/aports/main/qemu/src/qemu-4.0.0/rules.mak:69: ui/sdl2-gl.o] Error 1
In file included from /home/ncopa/aports/main/qemu/src/qemu-4.0.0/include/ui/sdl2.h:8,
                 from /home/ncopa/aports/main/qemu/src/qemu-4.0.0/ui/sdl2-input.c:30:
/usr/include/SDL2/SDL_syswm.h:77:10: fatal error: directfb.h: No such file or directory
 #include <directfb.h>
          ^~~~~~~~~~~~
compilation terminated.
make: *** [/home/ncopa/aports/main/qemu/src/qemu-4.0.0/rules.mak:69: ui/sdl2-input.o] Error 1
In file included from /home/ncopa/aports/main/qemu/src/qemu-4.0.0/include/ui/sdl2.h:8,
                 from /home/ncopa/aports/main/qemu/src/qemu-4.0.0/ui/sdl2.c:30:
/usr/include/SDL2/SDL_syswm.h:77:10: fatal error: directfb.h: No such file or directory
 #include <directfb.h>
          ^~~~~~~~~~~~
compilation terminated.
make: *** [/home/ncopa/aports/main/qemu/src/qemu-4.0.0/rules.mak:69: ui/sdl2.o] Error 1
```

**Comment by PureTryOut** [comment]
That is interesting, as DirectFB is still enabled as before...

**Comment by ncopa** [comment]
the autotools configure script sets `SDL_CFLAGS` which are included in the `sdl2.pc`. cmake does not do that. Apparently there are not many other users that uses both directfb and cmake.

I have a fix for it.

**Pull request closed without merging**